### PR TITLE
chore: allow explicitly provided prevarbifiedlist and newarbifiedlist path arguments when arbifying

### PR DESCRIPTION
**Pull request by Jason-Wanxt**
No need to add `newArbifiedList` to update because it will just update the former file and no new file will generate.

### CODE REVIEW

File: src/lib/store.ts
Comment (chrstph-dvx) [inline src/lib/store.ts:41]: leftover comment

File: src/lib/store.ts
Comment (chrstph-dvx) [inline src/lib/store.ts:42]: `const path = arbifiedList ?? getPath(l1ListName);`

File: src/commands/arbify.ts
Comment (chrstph-dvx) [inline src/commands/arbify.ts:22]: `const path = argvs.newArbifiedList ?? getPath(l1ListName);`

**Review by chrstph-dvx — COMMENTED** [review]
LGTM, just some small comments.

Also if you could run prettier please

File: src/lib/token_list_gen.ts
Comment (chrstph-dvx) [inline src/lib/token_list_gen.ts:428]: Sorry, I overlooked it on the first pass. Can we use `TOKENLIST_DIR_PATH` instead?
```typescript
const path = prevArbifiedList ?? TOKENLIST_DIR_PATH + listNameToFileName(arbTokenList.name);
```

**Review by chrstph-dvx — COMMENTED** [review]
Just one small fix to go

**Review by chrstph-dvx — APPROVED** [review]
Good job!

**Review by chrstph-dvx — APPROVED** [review]

**Review by DZGoldman — CHANGES_REQUESTED** [review]
awesome! let's get https://github.com/OffchainLabs/arb-token-lists/pull/69 merged + on this branch before we merge this one, so we can see it tested.

Also can we update the `update_all` tasks to use these new args / see it in action?

File: src/commands/update.ts
Comment (dewanshparashar) [inline src/commands/update.ts]: Are we sure we're keeping this log?

**Comment by Jason-Wanxt** [comment]
This should be removed in this [commit](
https://github.com/OffchainLabs/arb-token-lists/pull/75/commits/37eae187ea5d484ebae2875767e320637b3f5a33
)

Dewansh ***@***.***>于2023年2月6日 周一19:57写道：

File: update_all
Comment (chrstph-dvx) [inline update_all:12]: Can we update the CI too, so we generate them with the proper name?

**Review by chrstph-dvx — CHANGES_REQUESTED** [review]
LGTM, just one more change

File: update_all
Comment (Jason-Wanxt) [reply update_all:12]: Sure

**Review by chrstph-dvx — APPROVED** [review]

**Review by dewanshparashar — APPROVED** [review]

**Review by DZGoldman — APPROVED** [review]
looks good! 

Part of me thinks we should just get rid of the `getPath = (l1ListName: string)` method entirely, as it's kind of inherently ill-advised and cursed. IOW: --newArbifiedList would just always be required, and --prevArbifiedList would be optional / just produce a list with a version number of 1 if null. (Maybe an explicit --prevArbifiedList="NO_PREV" should be required so this doesn't become footgun).

Approving b/c it's mergable as is, if you prefer to merge and further the discussion/refactor in a new PR 👍

**Comment by Jason-Wanxt** [comment]
Sure, merged and will solve this in the next pr.

**Pull request merged**
